docs/developer/architecture/coding-principles.md

```diff
@@ -26,27 +26,67 @@
-
-### IV. Testing
+2. Avoid needlessly wrapping objects, especially primitive datatypes.
+
+### IV. General Coding Style
+
+1. Use `var` instead of explicit types (helps with clarity)
+2. Avoid `final` in method args and local variables
+3. Use `final` in field declarations
+4. Avoid `static` fields except in constants or when absolutely necessary. (you should be able to provide a reason).
+5. Use interfaces to define shared constants
+6. Use "minimally required types" (or "smallest possible API"), e.g. use `ObjectMapper` instead of `TypeManager`
+   , or use a `String` instead of a more complex object containing the String, etc.
+7. Use either `public` members, which are documented and tested, or `private` members.
+8. Avoid package-private members, especially if only needed for testing
+9. Avoid `protected` members unless they're intended to be overridden.
+10. Use package-private classes if they're not needed outside the package, e.g. implementation classes
+11. Avoid using `enum`s for anything other than named integer enumerations.
+12. Avoid using static classes as much as possible. Exceptions to this are helper functions and test utils, etc. as well as static inner classes.
+13. Use only camel case and no prefixes for naming.
+14. Avoid unnecessary `this.` (exception: in builders)
+15. Use static imports, as long as it is still clear what's being done. For example,
+    use `of(item1, item2).map(it -> it.someOperation)...` instead of `Stream.of(item1, item2)`, or `format("...",arg1)`
+    instead of `String.format("...", arg1)`. Helps with clarity and readability.
+16. Avoid `Optional` as method return type or method argument, except when designing a fluent API. Use `null` in signatures.
+17. Avoid cryptic variable names, especially in long methods. Instead, try to write them out, at least to a reasonable
+    extent.
+
+### V. Testing
 
 1. All handlers and services should have dedicated unit tests with mocks used for dependencies.
-2. When appropriate, prefer composing services via the constructor so that dependencies can be mocked as opposed to
-   instantiating dependencies directly, see `ProvisionManagerImpl`.
-
-### V. Data Objects
-
-1. Do not create constructors that take many parameters; instead use the builder pattern. An example
-   is `TransferProcess`.
-2. Note that the motivation behind use of builders is not for immutability (although that may be good in certain
+2. Prefer unit tests over all other test types: unit > integration/component > e2e
+3. When appropriate, prefer composing services via the constructor so that dependencies can be mocked as opposed to
+   instantiating dependencies directly.
+4. Use classes with static test functions to provide common helper methods, e.g. to instantiate an object.
+5. Use `[METHOD]_when[CONDITION]_should[EXPECTATION]` as naming template for test methods,
+   e.g. `verifyInput_whenNull_throwsNpe()` as opposed to `testInputNull()`
```

Comment:
Shouldn't this be `verifyInput_whenNull_shouldThrowNpe()`to match the pattern?

Reply:
ha, you're right, thanks!